Fix duplicate code in `PlatformTarget` subclasses, fix assets being included twice.

player-03 opened the pull request:
Having eight separate implementations made it harder to maintain.

When an asset embedding bug got fixed on desktop targets, it got overlooked on others. This lead to assets being included twice on those other targets: once embedded, and once normally. (#1898) Now, that behavior is controlled from one place, and the bug is fixed for eight targets at once.

This also standardizes the case of `asset.embed == true && asset.type == TEMPLATE`. Previously, some targets would prioritze `embed` over `type`, embedding the template into the app without even processing it as a template. Now, it will always prioritize `type`, processing templates as templates and never trying to embed them.

I left out `AIRPlatform`, `FlashPlatform`, and `HTML5Platform` since they have more complex asset embedding behavior. I haven't checked if the bug is present on any of those.

player-03 committed:
Unify asset copying code, and standardize asset embedding.

Having eight separate implementations made it harder to maintain. While an asset embedding bug got fixed on desktop targets, it got overlooked on others. This lead to assets being included twice on those other targets: once embedded, and once normally. (#1898) Now, that behavior is controlled from one place, and the bug is fixed for eight targets at once.

This also standardizes the case of `asset.embed == true && asset.type == TEMPLATE`. Previously, some targets would prioritze `embed` over `type`, embedding the template into the app without even processing it as a template. Now, it will always prioritize `type`, processing templates as templates and never trying to embed them.

Three targets are left out since they have more complex asset embedding behavior. I haven't checked if the bug is present on any of those.

player-03 committed:
Handle appinfo.json as a template on Tizen.

The old code appeared to be trying to do this, but it didn't work. When it found appinfo.json, it would call `copyAsset()` with an extra `context` argument, which is what you do for templates. However, `copyAsset()` only processes templates when `type == TEMPLATE`, otherwise it ignores the extra argument and processes it as a normal asset.

player-03 commented:
@MAJigsaw77 Check this out and let me know if it works.

player-03 committed:
Unify `PlatformTarget.watch()`.

The implementation was copy-pasted, so it makes more sense to have only a single copy. `HTML5Platform` added a comment about possible future changes, so I kept that. Future changes will still be possible by overriding the function.

player-03 committed:
Unify `PlatformTarget.clean()`.

All of the implementations were identical, except for `FlashPlatform`, which was identical with extra steps.

player-03 commented:
Cleaned up some more duplicate code while I was at it. I think that's it, though; the rest of the functions look unique.

player-03 commented on tools/platforms/TizenPlatform.hx:
```diff
@@ -258,31 +239,20 @@ class TizenPlatform extends PlatformTarget
-			else
-			{
-				AssetHelper.copyAsset(asset, path, context);
+				asset.type = AssetType.TEMPLATE;
```
_appinfo.json_ has been a special case since _TizenPlatform.hx_ was first created, and no one seems to have noticed it not working. Clearly the intent was to process it as a template, which is what I've done here. But... no one seems to have noticed the problem in seven years, so maybe we don't need the special case?

https://github.com/openfl/lime/blob/ecd7876c4b38853022e368415288ec2b93509ed1/tools/platforms/TizenPlatform.hx#L197-L209

player-03 commented on tools/platforms/WebAssemblyPlatform.hx:
```diff
@@ -481,42 +462,20 @@ class WebAssemblyPlatform extends PlatformTarget
 
 	@ignore public override function uninstall():Void {}
+
+	@ignore public override function watch():Void {}
```
`watch()` relies on `getDisplayHXML()`, which is missing from `AIRPlatform` and `TizenPlatform`, so that explains why we don't have it there.

However, `WebAssemblyPlatform` has `getDisplayHXML()` and doesn't have `watch()`. I don't know why, but I kept that behavior anyway. It seems like it ought to work, if we just enable it.

MAJigsaw77 commented:
<img width="261" height="280" alt="{0983020E-7020-487A-8788-454710077708}" src="https://github.com/user-attachments/assets/0cfa4582-7315-4d6e-8140-11ff9012316f" />

![Screenshot_20250901_093432](https://github.com/user-attachments/assets/a694c9b4-b27d-47d4-a2fc-cbbfaa11329b)

Can confirm that it works on `Android`.

player-03 committed:
Remove _appinfo.json_ special case.

Two reasons:

1. This is how it always worked in practice. The old special case actually did nothing due to an oversight.
2. Per the principle of least astonishment, Lime should avoid messing with the user's assets. If they wanted it to be a template, they would have specified `type="template"`.

player-03 committed:
Add a reminder to investigate `WebAssemblyPlatform.watch()`.

player-03 committed:
Merge branch '8.4.0-dev' into PlatformTarget_update_consistency
